src/daemon/ca/keys.rs

```diff
@@ -97,26 +96,42 @@ impl CertifiedKey {
 
         let not_after = self.incoming_cert().cert().validity().not_after();
 
-        let not_after = not_after.timestamp_millis();
-        let new_not_after = new_not_after.timestamp_millis();
+        let now = Time::now().timestamp_millis();
+        let until_not_after_millis = not_after.timestamp_millis() - now;
+        let until_new_not_after_millis = new_not_after.timestamp_millis() - now;
 
-        if not_after == new_not_after {
-            trace!("No change in not after time for certificate for key '{}'", self.key_id);
+        if until_new_not_after_millis < 0 {
+            // New not after time is in the past!
+            //
+            // This is rather odd. The parent should just exclude the resource class in the
+            // eligible entitlements instead. So, we will essentially just ignore this until
+            // they do.
+            false
```

Comment:
If this is odd, should a warning be logged?

Reply:
Yes, you're right. Pushed.

Still not logging in cases where we decide that there is no action required. No resource changes and not_after time unchanged. I could add debug / trace logging for this. What do you think?

Reply:
asking the question is answering it I guess.. pushed a 'debug' message in case there are no changes in resources or not after time.